### ISSUE TITLE
MethodError: isvalid(::String31)

### ISSUE THREAD

**Issue by jariji** [JuliaStrings/InlineStrings.jl#60]
This method would be useful:
```
ERROR: MethodError: no method matching isvalid(::String31)
```

**Comment by nickrobinson251** [comment]
i wonder if the correct definition is just the same as for `String` 
https://github.com/JuliaLang/julia/blob/cdcb07f13c10611f3a2fe4d491a8d0342ad11d12/base/strings/string.jl#L194-L201

(here's the `u8_isvalid` definition: https://github.com/JuliaLang/julia/blob/cdcb07f13c10611f3a2fe4d491a8d0342ad11d12/src/support/utf8.c#L542-L596)

i.e. something like
```julia
Base.isvalid(s::InlineString) = ccall(:u8_isvalid, Int32, (Ptr{UInt8}, Int), s, sizeof(s)) != 0
```

**Comment by quinnj** [comment]
Nah, we don't need to check the length; yeah, that definition sounds good to me. @jariji, mind making a PR with a test?

**Comment by stevengj** [comment]
With https://github.com/JuliaLang/julia/pull/47880, you should just be able to do:
```jl
Base.isvalid(s::InlineString) = isvalid(String, s)
```

**Comment by KristofferC** [comment]
I think it is fine to close this with the feature being included into 1.10 which will soon be the new LTS:
